pkg/trace/traceutil/processed_trace.go

```diff
@@ -18,3 +18,36 @@ type ProcessedTrace struct {
 	TracerHostname         string
 	ClientDroppedP0sWeight float64
 }
+
+// Clone creates a copy of ProcessedTrace, cloning p, p.TraceChunk, and p.Root. This means it is
+// safe to modify the returned ProcessedTrace's (pt's) fields along with fields in
+// pt.TraceChunk and fields in pt.Root.
+//
+// The most important consequence of this is that the TraceChunk's Spans field can be assigned,
+// *BUT* the Spans value itself should not be modified. i.e. This is ok:
+//
+//	pt2 := pt.Clone()
+//	pt2.TraceChunk.Spans = make([]*pb.Span)
+//
+// but this is NOT ok:
+//
+//	pt2 := pt.Clone()
+//	pt2.TraceChunk.Spans[0] = &pb.Span{} // This will be visible in pt.
+func (pt *ProcessedTrace) Clone() *ProcessedTrace {
```

Comment:
it probably wouldn't hurt to have a little unit test for this. i.e. one that makes a clone, then changes some fields in it and makes sure the original doesn't change. We're doing that a bit in the other tests, but having it here makes it more explicit where the problem is should we introduce a bug in the future.